docgenerator/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+find_package(Doxygen REQUIRED)
+find_package(Sphinx REQUIRED)
+
+# adapted from https://devblogs.microsoft.com/cppblog/clear-functional-c-documentation-with-sphinx-breathe-doxygen-cmake/
+# Find all headers
+file(GLOB_RECURSE HEADERS ${PROJECT_SOURCE_DIR}/*.hxx)
+
+set(DOXYGEN_INPUT_DIR ${PROJECT_SOURCE_DIR}/)
+set(DOXYGEN_OUTPUT_DIR ${PROJECT_SOURCE_DIR}/docs/doxygen/)
+set(DOXYGEN_INDEX_FILE ${DOXYGEN_OUTPUT_DIR}/html/index.html)
+set(DOXYFILE_IN ${PROJECT_SOURCE_DIR}/docgenerator/scripts/Doxyfile.in)
+set(DOXYFILE_OUT ${PROJECT_SOURCE_DIR}/docgenerator/scripts/Doxyfile)
+
+#Replace variables inside @@ with the current values
+configure_file(${DOXYFILE_IN} ${DOXYFILE_OUT} @ONLY)
+
+file(MAKE_DIRECTORY ${DOXYGEN_OUTPUT_DIR}) #Doxygen won't create this for us
+add_custom_command(OUTPUT ${DOXYGEN_INDEX_FILE}
```

Comment:
All of these targets are at build time, so when calling `make`, not `cmake` (regarding the comment above).